tests/installation/hostname_inst.pm

```diff
@@ -18,18 +18,22 @@ use base 'y2_installbase';
 use strict;
 use warnings;
 use testapi;
+use version_utils;
 
 sub run {
     assert_screen "before-package-selection";
     select_console 'install-shell';
-    if (my $expected_install_hostname = get_var('EXPECTED_INSTALL_HOSTNAME')) {
-        # EXPECTED_INSTALL_HOSTNAME contains expected hostname YaST installer
-        # got from environment (DHCP, 'hostname=' as a kernel cmd line argument
-        assert_script_run "test \"\$(hostname)\" == \"$expected_install_hostname\"";
-    }
-    else {
-        # 'install' is the default hostname if no hostname is get from environment
-        assert_script_run 'test "$(hostname)" == "install"';
+    # NICTYPE_USER_OPTIONS="hostname=myguest" causes a fake DHCP hostname provided to SUT
+    my $NICTYPE_USER_OPTIONS      = get_required_var('NICTYPE_USER_OPTIONS');
+    my $expected_install_hostname = ($NICTYPE_USER_OPTIONS =~ s/hostname=//r);
+    # Before SLE15-SP2, yast didn't take during installation the hostname by DHCP
+    # See fate#319639
+    # 'install' is the default hostname if no hostname is get from environment
+    $expected_install_hostname = 'install' if (is_sle('<15-SP2'));
+    if (is_sle('<15-SP2') && (script_run(qq{test "\$(hostname)" == "linux"}) == 0)) {
+        record_soft_failure('bsc#1166778 - Default hostname in SLE15-SP1 is not "install"');
+    } else {
+        assert_script_run(qq{test "\$(hostname)" == "$expected_install_hostname"});
```

Comment:
I'd `grep -q /etc/hostname $expected_install_hostname` instead, taking into account missing keys and stuff :), but test also works

Reply:
In the case of 'linux' grep will return 0 when the hostname is linux-blah

Reply:
I suppose `grep "^${expected_install_hostname}$" /etc/hostname` should do.

I would not use -q, since it will make easier for reviewer to see grep output.

Reply:
But

```bash
sergio@sergio-latitude:~$ echo 'grep "^${expected_install_hostname}$"' /etc/hostname | wc -c
52
sergio@sergio-latitude:~$ echo 'test "$(hostname)" == "$expected_install_hostname"' | wc -c
51
```